From 2e320484e26be7677fefa66860ff2c1a9fe8e598 Mon Sep 17 00:00:00 2001
From: Patrice Brissette <patricebrissette@gmail.com>
Date: Tue, 5 May 2026 17:06:55 -0400
Subject: [PATCH] zebra: ARP/ND Suppression disabled for kernel mac ext learn

ARP/ND suppression is not supported in kernel MAC external learn mode.

Signed-off-by: Patrice Brissette <pbrisset@cisco.com>
Signed-off-by: Tamer Ahmed <tamerahmed@microsoft.com>
(cherry picked from commit db74c19c0515e7390e53ee71e8dcad5be68a1ed5)
---
 zebra/zebra_evpn_neigh.c | 14 +++++++++++---
 1 file changed, 11 insertions(+), 3 deletions(-)

diff --git a/zebra/zebra_evpn_neigh.c b/zebra/zebra_evpn_neigh.c
index 69870b88fe..b7bbd97c35 100644
--- a/zebra/zebra_evpn_neigh.c
+++ b/zebra/zebra_evpn_neigh.c
@@ -112,7 +112,11 @@ int zebra_evpn_rem_neigh_install(struct zebra_evpn *zevpn,
 		flags |= DPLANE_NTF_ROUTER;
 	ZEBRA_NEIGH_SET_ACTIVE(n);
 
-	dplane_rem_neigh_add(vlan_if, &n->ip, &n->emac, flags, was_static);
+	/* Skip Installation to Kernel, in mac external learn mode
+	 * ARP/ND Suppression/Asymmetric IRB is not supported in MAC External learn mode
+	 */
+	if (!zebra_mac_ext_learn_mode())
+		dplane_rem_neigh_add(vlan_if, &n->ip, &n->emac, flags, was_static);
 
 	return ret;
 }
@@ -812,8 +816,12 @@ static int zebra_evpn_neigh_uninstall(struct zebra_evpn *zevpn,
 
 	ZEBRA_NEIGH_SET_INACTIVE(n);
 	n->loc_seq = 0;
-
-	dplane_rem_neigh_delete(vlan_if, &n->ip);
+	/* Installation to Kernel is skipped, in mac external learn mode
+	 * Hence no uninstall required
+	 * ARP/ND Suppression/Asymmetric IRB is not supported in MAC External learn mode
+	 */
+	if (!zebra_mac_ext_learn_mode())
+		dplane_rem_neigh_delete(vlan_if, &n->ip);
 
 	return 0;
 }
-- 
2.34.1

